llama/runner/runner.go

```diff
@@ -451,14 +451,27 @@ func (s *Server) processBatch(tokenBatch *llama.Batch, embedBatch *llama.Batch)
 		sequence := strings.Join(seq.pendingResponses, "")
 
 		if ok, stop := findStop(sequence, seq.stop); ok {
-			slog.Debug("hit stop token", "stop", seq.stop)
-
-			trimCacheLen := len(seq.pendingResponses) - 1
-			seq.pendingResponses = truncateStop(seq.pendingResponses, stop)
-			trimCacheLen -= len(seq.pendingResponses)
+			slog.Debug("hit stop token", "pending", seq.pendingResponses, "stop", stop)
+
+			var truncatedToken bool
+			origLen := len(seq.pendingResponses)
+			seq.pendingResponses, truncatedToken = truncateStop(seq.pendingResponses, stop)
```

Comment:
```suggestion
			seq.pendingResponses, truncated = truncateStop(seq.pendingResponses, stop)
```

`truncated` might be more `bool`-y

Reply:
or `found` similar to `Cut` etc in the stdlib: https://pkg.go.dev/strings#Cut

Reply:
I actually originally named it `truncated` but I thought that it was a bit misleading - this value specifically indicates whether an individual token was truncated, rather than the overall sequence of strings being truncated. `found` is also not quite right - that suggests that the bool is whether the stop string was located (which is what it means in the std library). That actually should always be true unless there is a bug.

So even though `truncatedToken` is a little awkward and verbose I think it is more accurately descriptive. I might reverse it to be `tokenTruncated` though.